src/fs/feature/xattr.rs

```diff
@@ -50,59 +50,77 @@ pub enum FollowSymlinks {
 #[derive(Debug, Clone)]
 pub struct Attribute {
     pub name: String,
-    pub size: usize,
+    pub value: String,
 }
 
 
 #[cfg(any(target_os = "macos", target_os = "linux"))]
 pub fn list_attrs(lister: &lister::Lister, path: &Path) -> io::Result<Vec<Attribute>> {
     use std::ffi::CString;
 
-    let c_path = match path.to_str().and_then(|s| CString::new(s).ok()) {
-        Some(cstring) => cstring,
+    let c_path = match path.to_str().map(CString::new) {
+        Some(cstring) => match cstring {
+            Ok(cstring) => cstring,
+            Err(e) => {
+                return Err(io::Error::new(io::ErrorKind::Other, e));
+            }
+        },
```

Comment:
Using `cstring` for two different values is very confusing.

```suggestion
        Some(cstring_result) => match cstring_result {
            Ok(cstring) => cstring,
            Err(e) => {
                return Err(io::Error::new(io::ErrorKind::Other, e));
            }
        },
```